### ISSUE TITLE
Pydantic serializer warnings: Expected `enum` but got `str` - serialized value may not be as expected

### ISSUE THREAD

**Issue by jwmatthews** [konveyor/kai#240]
I see occasional messages like below in log output from backend when running against coolstore

```
kai_1               | INFO - 2024-07-18 21:11:21,643 - kai.server - [           server.py:219  - get_incident_solutions_for_file()] - START - App: 'coolstore', File: 'src/main/java/com/redhat/coolstore/model/InventoryEntity.java' with 6 incidents'
kai_1               | /opt/app-root/lib64/python3.12/site-packages/pydantic/main.py:398: UserWarning: Pydantic serializer warnings:
kai_1               |   Expected `enum` but got `str` - serialized value may not be as expected
kai_1               |   return self.__pydantic_serializer__.to_json(
kai_1               | /opt/app-root/lib64/python3.12/site-packages/pydantic/main.py:398: UserWarning: Pydantic serializer warnings:
kai_1               |   Expected `enum` but got `str` - serialized value may not be as expected
kai_1               |   return self.__pydantic_serializer__.to_json(
kai_1               | /opt/app-root/lib64/python3.12/site-packages/pydantic/main.py:398: UserWarning: Pydantic serializer warnings:
kai_1               |   Expected `enum` but got `str` - serialized value may not be as expected
kai_1               |   return self.__pydantic_serializer__.to_json(
```

```
kai_1               | INFO - 2024-07-18 21:13:01,672 - kai.server - [           server.py:219  - get_incident_solutions_for_file()] - START - App: 'coolstore', File: 'src/main/java/com/redhat/coolstore/rest/OrderEndpoint.java' with 8 incidents'
kai_1               | /opt/app-root/lib64/python3.12/site-packages/pydantic/main.py:398: UserWarning: Pydantic serializer warnings:
kai_1               |   Expected `enum` but got `str` - serialized value may not be as expected
kai_1               |   return self.__pydantic_serializer__.to_json(
```

```
kai_1               | INFO - 2024-07-18 21:13:45,441 - kai.server - [           server.py:219  - get_incident_solutions_for_file()] - START - App: 'coolstore', File: 'src/main/java/com/redhat/coolstore/rest/RestApplication.java' with 4 incidents'
kai_1               | /opt/app-root/lib64/python3.12/site-packages/pydantic/main.py:398: UserWarning: Pydantic serializer warnings:
kai_1               |   Expected `enum` but got `str` - serialized value may not be as expected
kai_1               |   return self.__pydantic_serializer__.to_json(
```

**Comment by JonahSussman** [comment]
Should be fixed by https://github.com/konveyor/kai/pull/232. Pydantic doesn't like Python's raw `Enum` stuff when using strings. Converting our `Enum`s to `StrEnum`s should fix it.

**Comment by jwmatthews** [comment]
I see below from run_demo.py output in PR #232 

```
INFO - 2024-07-24 08:06:09,830 - __main__ - [         run_demo.py:173  -         process_file()] - File #8 of 26 - Processing src/main/java/com/redhat/coolstore/model/ShoppingCart.java which has 1 incidents.
/Users/jmatthews/git/jwmatthews/kai/env/lib/python3.12/site-packages/pydantic/main.py:415: UserWarning: Pydantic serializer warnings:
  Expected `enum` but got `str` - serialized value may not be as expected
  return self.__pydantic_serializer__.to_json(
```

**Comment by JonahSussman** [comment]
Should be fixed by e1888ebf4a2bc482971aaf64cd243a1f41a529f1. The type for `batch_mode` is a `BatchMode`, which is an enum, but I set the default value as a `str`, producing the warning.
